libraries/azure-app-configuration-importer/src/internal/parsers/defaultConfigurationSettingsConverter.ts

```diff
@@ -36,39 +38,64 @@ export class DefaultConfigurationSettingsConverter implements ConfigurationSetti
     let configurationSettings = new Array<
       SetConfigurationSettingParam<string | FeatureFlagValue>
     >();
+
     let featureFlagsConfigSettings = new Array<SetConfigurationSettingParam<FeatureFlagValue>>();
-    let foundFeatureManagement = false;
-    let featureFlags: any = {};
-    let featureManagementKeyWord = "";
-    let enabledForKeyWord = "";
+    let foundMsFmSchema = false;
+    let foundLegacySchema = false;
+    let legacySchemaFeatureManagementKeyWord = "";
+
+    const featureFlagsDict: any = {};
 
     if (options.format == ConfigurationFormat.Properties && !options.skipFeatureFlags) {
       this.checkFeatureManagementExist(config);
     }
 
-    for (let i = 0; i < Constants.FeatureManagementKeyWords.length; i++) {
+    for (let i = 0; i < Constants.FeatureManagementKeyWords.length - 1; i++) {
       if (Constants.FeatureManagementKeyWords[i] in config) {
-        if (foundFeatureManagement) {
+        if (foundLegacySchema) {
           throw new ArgumentError(
-            "Unable to proceed because data contains multiple sections corresponding to Feature Management."
+            `Unable to proceed because data contains multiple sections corresponding to Feature Management. with the key, ${Constants.FeatureManagementKeyWords[i]}`
           );
         }
+        
+        foundLegacySchema = true;
+        legacySchemaFeatureManagementKeyWord = Constants.FeatureManagementKeyWords[i];
 
-        featureManagementKeyWord = Constants.FeatureManagementKeyWords[i];
-        enabledForKeyWord = Constants.EnabledForKeyWords[i];
-        const featureManagementKey = featureManagementKeyWord as keyof object;
-        featureFlags = config[featureManagementKey];
-        delete config[featureManagementKey];
-        foundFeatureManagement = true;
+        const legacyFeatureManagementKey = legacySchemaFeatureManagementKeyWord as keyof object;
+        featureFlagsDict[legacyFeatureManagementKey] = config[legacyFeatureManagementKey];
+        delete config[legacyFeatureManagementKey];
       }
     }
 
-    if (foundFeatureManagement && !options.skipFeatureFlags) {
+    const msFeatureManagementKeyWord = Constants.FeatureManagementKeyWords[3];
+    if (msFeatureManagementKeyWord in config) {
+      if (foundLegacySchema &&
+        Object.keys(config[msFeatureManagementKeyWord as keyof object]).some(key => key !== Constants.FeatureFlagsKeyWord)) {
+        throw new ArgumentError(
+          `Unable to proceed because data contains an already defined section with same schema corresponding to Feature Management with the key, ${legacySchemaFeatureManagementKeyWord}.`
```

Comment:
I know this error use the same words as above, but it's quite confusion-leading here.

Reply:
I have updated to `Unable to proceed because data contains an already defined legacy schema section with the key, ${legacySchemaFeatureManagementKeyWord}.` not sure if its better. Let me know what you think